gcp/src/main/java/org/apache/iceberg/gcp/biglake/BigLakeCatalog.java

```diff
@@ -0,0 +1,393 @@
+    }
+
+    return true;
+  }
+
+  @Override
+  public boolean setProperties(Namespace namespace, Map<String, String> properties) {
+    HiveDatabaseOptions.Builder optionsBuilder =
+        loadDatabase(namespace).toBuilder().getHiveOptionsBuilder();
+    properties.forEach(optionsBuilder::putParameters);
+    bigLakeClient.updateDatabaseParameters(
+        databaseName(namespace), optionsBuilder.getParametersMap());
+    return true;
+  }
+
+  @Override
+  public boolean removeProperties(Namespace namespace, Set<String> properties) {
+    HiveDatabaseOptions.Builder optionsBuilder =
+        loadDatabase(namespace).toBuilder().getHiveOptionsBuilder();
+    properties.forEach(optionsBuilder::removeParameters);
+    bigLakeClient.updateDatabaseParameters(
+        databaseName(namespace), optionsBuilder.getParametersMap());
+    return true;
+  }
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(Namespace namespace) {
+    if (namespace.isEmpty()) {
+      // Calls getCatalog to check existence. BLMS catalog has no metadata today.
+      bigLakeClient.getCatalog(catalogName);
+      return ImmutableMap.of();
+    } else if (namespace.levels().length == 1) {
+      return metadata(loadDatabase(namespace));
+    } else {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+  }
+
+  @Override
+  public String name() {
+    return name;
+  }
+
+  @Override
+  protected Map<String, String> properties() {
+    return bigLakeProperties == null ? ImmutableMap.of() : bigLakeProperties;
+  }
+
+  @Override
+  public void setConf(Object conf) {
+    this.conf = conf;
+  }
+
+  @Override
+  public void close() throws IOException {
+    if (closeableGroup != null) {
+      closeableGroup.close();
+    }
+  }
+
+  @Override
+  protected boolean isValidIdentifier(TableIdentifier tableIdentifier) {
+    return tableIdentifier.namespace().levels().length == 1;
+  }
+
+  private String databaseLocation(String dbId) {
+    String warehouseLocation =
+        LocationUtil.stripTrailingSlash(
+            bigLakeProperties.get(CatalogProperties.WAREHOUSE_LOCATION));
+    Preconditions.checkNotNull(warehouseLocation, "Data warehouse location is not set");
+    return String.format("%s/%s.db", LocationUtil.stripTrailingSlash(warehouseLocation), dbId);
+  }
+
+  private static TableIdentifier tableIdentifier(Table table) {
+    TableName tableName = TableName.parse(table.getName());
+    return TableIdentifier.of(Namespace.of(tableName.getDatabase()), tableName.getTable());
+  }
+
+  private static Namespace namespace(Database db) {
+    return Namespace.of(DatabaseName.parse(db.getName()).getDatabase());
+  }
+
+  private TableName tableName(String dbId, String tableId) {
+    return TableName.of(bigLakeProjectId, bigLakeRegion, catalogId, dbId, tableId);
+  }
+
+  private String databaseId(Namespace namespace) {
+    if (namespace.levels().length != 1) {
+      throw new NoSuchNamespaceException(
```

Comment:
This isn't the right place to throw `NoSuchNamespaceException` because there's no context for the operation that called this. I think this should return either `level(0)` or `null` and let the caller throw exceptions.

Reply:
Done.